pkg/service/audit_utils.go

```diff
@@ -389,6 +391,57 @@ func CreateApiKeyAuditEvent(ctx context.Context, al *zap.Logger, action string,
 	}
 }
 
+func GenerateApiKeyAuditEvent(ctx context.Context, al *zap.Logger, action string, id string) {
+	sd, ok := GetSessionDataFromContext(ctx)
+	if !ok {
+		_log.Warn("unable to create audit event: could not fetch info from context")
+		return
+	}
+
+	detail := &audit.EventDetail{
+		Message: fmt.Sprintf("ApiKey %sd for user %s", action, id),
+		Meta: map[string]string{
+			"apikey": id,
+		},
+	}
+	if err := audit.CreateV1Event(al, sd, detail, fmt.Sprintf("apikey.%s.success", action), ""); err != nil {
+		_log.Warn("unable to create audit event", err)
+	}
+}
+
+func DownloadKubeconfigAuditEvent(ctx context.Context, al *zap.Logger, user string) {
+	sd, ok := GetSessionDataFromContext(ctx)
+	if !ok {
+		_log.Warn("unable to create audit event: could not fetch info from context")
+		return
+	}
+
+	detail := &audit.EventDetail{
+		Message: fmt.Sprintf("Kubeconfig Downloaded for user %s", user),
+	}
+	_log.Infow(fmt.Sprintf("Kubeconfig Downloaded for user %s", user))
+	if err := audit.CreateV1Event(al, sd, detail, fmt.Sprintf("Kubeconfig Download success %s", user), ""); err != nil {
+		_log.Warn("unable to create audit event", err)
+	}
+	_log.Infow("Audit event created")
```

Comment:
this gets printed everytime even if due to whatever reason audit event does not get created.

Reply:
Will correct this